src/Automation/ReadMe.md

```diff
@@ -0,0 +1,65 @@
+﻿When looking at the workflow of this project, please start at
+`AutomationFactory.CreateAutomationSessionFactory`.
+
+## Compatibility practices
+
+The set of public interfaces defined in this project constitutes the
+requirements to run an AxeWindows automated scan of a given application
+and take action on the errors detected by the scan. We expect  the set
+of interfaces will grow as the requirements grow. Ideally, customers could
+update to new versions of the Axe.Windows package, taking advantage of rule
+improvements or updated element properties for example, without having to modify
+their code. To that end, the following sections describe types of
+changes to the interfaces.
+
+### Considerations
+
+Because this library is a strong-named .Net assembly, we can be sure 
+the code that depends on it will always link to the expected 
+version of the dll at runtime. The benefit is that we don't need 
```

Comment:
This statement is true if the top-level app (i.e., the app consuming axe-windows) uses strong naming. If, however, the top-level application does not use strong naming, then the strong naming in axe-windows will be ignored

Reply:
The real question is if the difference should impact the current design. Thoughts? If the consumer can dynamically link to an older or newer version of AxeWindows, is that something we need to plan for? How concerned to we need to be about this possibility?

Reply:
There *is* a scenario where a caller could get an unexpected version of axe-windows, but it's sufficiently convoluted that I don't think we need to optimize for it. I'm fine with the current compatibility strategy, but I guess I'm suggesting that we omit the reference to strong naming because our compatibility story is independent of strong naming.

---
In reply to: [291302963](https://github.com/microsoft/axe-windows/pull/46#discussion_r291302963) [](ancestors = 291302963)